app/ydoc-server/src/languageServerSession.ts

```diff
@@ -570,7 +587,23 @@ class ModulePersistence extends ObservableV2<{ removed: () => void }> {
           syncModule.applyEdit(edit)
         } else {
           const { root, spans } = Ast.parseModuleWithSpans(code, syncModule)
-          syncModule.syncRoot(root)
+
+          let rootToSync = root
+          const metadataSnapshot = metadata.ide.snapshot
+          if (metadataSnapshot && idMapJson) {
+            const snapshotCode = ModulePersistence.decodeCodeSnapshot(metadataSnapshot)
+            if (snapshotCode !== code || 1 === 1 /* simulate code mismatch */) {
+              const { root, spans } = Ast.parseModuleWithSpans(snapshotCode)
+              const edit = root.module.edit()
```

Comment:
It isn't necessary to use `edit()`--`edit()` creates a copy to provide semi-transactional semantics, but we will be committing these changes unconditionally so `syncModule` can be used directly in place of the `edit` value.